shoplist/src/main/java/com/netguru/codereview/ui/MainViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.netguru.codereview.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.netguru.codereview.network.ShopListApiMock
+import com.netguru.codereview.network.ShopListRepository
+import com.netguru.codereview.network.model.ShopListItemResponse
+import com.netguru.codereview.network.model.ShopListResponse
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+class MainViewModel : ViewModel() {
+
+    private val shopListRepository = ShopListRepository(ShopListApiMock())
+
+    val shopLists = MutableLiveData<List<Pair<ShopListResponse, List<ShopListItemResponse>>>>()
+    private val eventLiveData = MutableLiveData<String>()
+
+    init {
+        viewModelScope.launch {
+            val lists = shopListRepository.getShopLists()
+            val data = mutableListOf<Pair<ShopListResponse, List<ShopListItemResponse>>>()
+            for (list in lists) {
+                val items = shopListRepository.getShopListItems(list.list_id)
```

Comment:
u can't call a long-running operation on the main thread 
instead, run it on io or default 
suggestion :
```suggestion
          val items = withContext(Dispatchers.Default)
                {
                    shopListRepository.getShopListItems(list.list_id)
                }
```